common/cache/lru.go

```diff
@@ -242,28 +251,43 @@ func (c *lru) Size() int {
 	c.mut.Lock()
 	defer c.mut.Unlock()
 
+	c.evictExpiredItems()
+
 	return len(c.byKey)
 }
 
+// evictExpiredItems evicts all items in the cache which are expired
+func (c *lru) evictExpiredItems() {
+	if !c.activelyEvict {
+		return // do nothing if activelyEvict is not set
+	}
+
+	for elt := c.byAccess.Back(); elt != nil && c.isEntryExpired(elt.Value.(*entryImpl), c.now()); elt = c.byAccess.Back() {
```

Comment:
I still propose to simplify this for a reader:
```suggestion
	for elt := c.byAccess.Back(); elt != nil; elt = c.byAccess.Back() {
	if !c.isEntryExpired(elt.Value.(*entryImpl), c.now()) {
	    // List is sorted by item age, so we can stop as soon as we found first non expired item.
	    break
	}
```

Reply:
Sure, moved it around